### DISCUSSION TITLE
Gotenberg over HTTP/2

### DISCUSSION THREAD

**Discussion by Svendegroote91** [gotenberg/gotenberg#364] [Q&A] [answered]
Currently we are facing 413 responses from Cloud Run when we invoke Gotenberg over HTTP/1. 
I believe HTTP/2 is currently not supported, is this on the roadmap?

**Comment by gulien** [comment, marked as answer]
Hello @Svendegroote91 

Indeed, there is a limit of 32 MB request size if using HTTP/1 server on Cloud Run.
HTTP2 (h2/h2c?) is on my TODO list, but no ETA yet. 
I will gladly review a PR though 😄

**Reply by Svendegroote91** [reply]
Hi @gulien , 

Thanks for the swift response.
Ok that is good to know. 
Is this is a good first issue to tackle or is it rather advanced?  My go skills are a bit rusty 😄

**Reply by gulien** [reply]
Dunno to be honest 😄 Echo provides a [cookbook for HTTP2](https://echo.labstack.com/cookbook/http2/) but I would prefer HTTP2 without TLS ("h2c" AFAIK). Integration with Gotenberg may not be that hard though.

**Comment by KillianMeersman** [comment]
Hey @gulien

I opened #366 for h2c support. I've tested it on our system and it works as expected. Let me know what you think :smile: .
